### ISSUE TITLE
Support __call__ on metaclasses

### ISSUE THREAD

**Issue by ilevkivskyi** [python/mypy#14122]
Currently we don't handle `__call__()` on metaclasses correctly. For example:

```python
class M(type):
    def __call__(self, x: int) -> Any: ...

class C(metaclass=M): ...

C(x=1)
```
results in a false positive. Also similarly handling of metaclass `__call__()` in protocols is not very principled, see e.g. https://github.com/python/mypy/pull/14121

**Comment by cgarciae** [comment]
Hey! I am getting inconsistent results for pyright/pylance for this type that override `__call__`:

```python
class ContainerMetaclass(ABCMeta):
  def __call__(self, value: A, **metadata: tp.Any) -> A:
    ...
    
class Param(metaclass=ContainerMetaclass):
  ...
```
Slight modifications to the code change the type inferred for the output of the constructor:

```python
#        vvvvvvvv<---- type hint
kernel: jax.Array = nnx.initializers.lecun_normal()(
    ctx.make_rng("params"),
    (seq_len, seq_len, n_features, n_features),
    jnp.float32,
)
self.kernel = nnx.Param(kernel) # infers: jax.Array (correct)
```
```python
#  removed type hint
kernel = nnx.initializers.lecun_normal()(
    ctx.make_rng("params"),
    (seq_len, seq_len, n_features, n_features),
    jnp.float32,
)
self.kernel = nnx.Param(kernel) # infers: Param[jax.Array] (wrong)
```
